state/account_test.go

```diff
@@ -108,6 +108,12 @@ func TestBalance(t *testing.T) {
 	require.Equal(0, state.Balance.Cmp(big.NewInt(30)))
 	// Sub 40 to the balance
 	require.Equal(ErrNotEnoughBalance, state.SubBalance(big.NewInt(40)))
+
+	require.True(state.HasSufficientBalance(big.NewInt(30)))
+	require.False(state.HasSufficientBalance(big.NewInt(31)))
+
+	require.Contains(state.AddBalance(big.NewInt(-1)).Error(), ErrInvalidAmount.Error())
+	require.Contains(state.SubBalance(big.NewInt(-1)).Error(), ErrInvalidAmount.Error())
```

Comment:
1. test nil input will also cause error
2. for `SubBalance`, test try to subtract a larger amount also generates error

Reply:
1、added nil input check
2、done by https://github.com/iotexproject/iotex-core/blob/6823ba87dc0af38b918ed750f454694dfec4dbe2/state/account_test.go#L110